### PR TITLE
phone-number: Rewrite description and add tests

### PR DESCRIPTION
**Pull request by behrtam**
Removes the part about how to test, as we have other docs for that. Now it also describes the telephone number system in a bit more detail.

Changes some test that used an invalid number but assumed it was valid.

Adds two tests that check the range of 2-9 for the first digit of the area and the exchange code.

Adds a test with a valid telephone number with country code and punctuation.

### CODE REVIEW

**Comment by behrtam** [comment]
We could also add new properties, but I think that's maybe to much.

```json
{
  "description": "extract area code",
  "property": "area",
  "phrase": "(223) 456-7890",
  "expected": "223"
},
{
  "description": "extract exchange code",
  "property": "exchange",
  "phrase": "223.456.7890",
  "expected": "456"
},
```

**Comment by kotp** [comment]
I like the tests with the leading number avoiding the 1.  As far as the parts of the number, the names of those parts sometimes comes up in reviews (at least on the Ruby track).

File: exercises/phone-number/description.md [hunk 1]
```diff
@@ -1,16 +1,22 @@
...
 
-Hint: Only make one test pass at a time. Disable the others, then flip
-each on in turn after you get the current failing one to pass.
+The format is usually represented as
+```
+(NXX)-NXX-XXXX
+```
+where N is any digit from 2 through 9 and X is any digit from 0 through 9.
+
+You task is to clean up differently formated telephone numbers by removeing punctuatin and the country code (1) if present.
```
Comment (rpottsoh) [inline exercises/phone-number/description.md]: `You` should be `Your` on line 12.

**Comment by rpottsoh** [comment]
@behrtam LGTM!

**Comment by behrtam** [comment]
So, should I add the additional properties like `area code` or `exchange code` as proposed in the second comment or do we stick with the one existing property?

**Comment by rpottsoh** [comment]
@behrtam I am fine with adding those tests.  However, this [PR](https://github.com/exercism/x-common/pull/477) back in December, touched on the issue of extraction and pretty printing and the decision was made to leave those tests out of the canonical-data.json.

**Comment by behrtam** [comment]
@petertseng wrote:
> Pretty printing and extracting subparts of numbers: Yeah, doesn't seem that important to me, and the README doesn't mention it, so I would not add them to the JSON file. It would make sense if people agreed to add it and it got added to the README and JSON file together. But that can come later.

Well those two additional properties are mentioned in the new description.

**Comment by rpottsoh** [comment]
Good point.  It is later now.  :)


Sent from my android device.

File: exercises/phone-number/description.md (hunk 1 above)
Comment (Insti) [inline exercises/phone-number/description.md]: and "Punctuation" is misspelled.

Comment (Insti) [inline exercises/phone-number/description.md]: and "Removing"

**Comment by Insti** [comment]
>  those two additional properties are mentioned in the new description.

The conclusion in the previous PR was to not add them anyway.

I vote NO, to adding the new properties.

**Comment by behrtam** [comment]
I wouldn't really call a single post without interaction "conclusion" ...  as I don't have any opinion about the additionl properties I'm fine with not adding them.

I will fix the typos later when I'm back at a real device.

File: exercises/phone-number/description.md (hunk 1 above)
Comment (behrtam) [inline exercises/phone-number/description.md]: Thx!
Sorry, for the sloppy work. I'll try to remember to use a spell checker next time.

**Pull request merged**